CTFd/plugins/challenges/__init__.py

```diff
@@ -161,6 +182,8 @@ def fail(cls, user, team, challenge, request):
         :return:
         """
         data = request.form or request.get_json()
+        if not data or not data.get("submission"):
```

Comment:
@ColdHeat Need feedback on this - Should I insert a fail anyways if `data["submission"]` does not exist or is `None`? This change makes those inputs NOT add a row to to Fails